x/ibc/core/04-channel/keeper/handshake.go

```diff
@@ -147,6 +148,15 @@ func (k Keeper) ChanOpenTry(
 		)
 	}
 
+	// empty-string is a sentinel value for "allow any identifier" to be selected by
+	// the counterparty channel
+	if provedChannelID != channelID && provedChannelID != "" {
```

Comment:
nit, I know the result is the same but if we change the order the second clause won't need to be evaluated as the entire condition will return false. Usually, in JS you would use `if object && object.field...` to check for `undefined` objects.

```suggestion
	if provedChannelID != "" && provedChannelID != channelID  {
```

Reply:
yea I agree which is why I did this in `msgs.go`. I left how the spec ordered it in handshake for consistency. This is also how it is in connection handshake so I'd prefer to update this in a separate pr since it isn't api breaking

Reply:
cc/ @cwgoes should we apply this to the spec? I think it reads slightly better

Reply:
Applied to the spec in https://github.com/cosmos/ics/commit/890e3d0c534ed46b97c794fa0947022380815dbe